File: android/app/src/main/java/com/alamada/childmonitor/models/Enrollment.java
package com.alamada.childmonitor.models;

import com.google.gson.annotations.SerializedName;

public class Enrollment {
    @SerializedName("id")              public int    id;
    @SerializedName("child_id")        public int    childId;
    @SerializedName("enrollment_date") public String enrollmentDate;
    @SerializedName("school_year")     public int    schoolYear;
    @SerializedName("status")          public String status;
    @SerializedName("remarks")         public String remarks;
    @SerializedName("child")           public Child  child;
}

File: backend/app/Models/Enrollment.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Enrollment extends Model
{
    use HasFactory;

    protected $fillable = [
        'child_id',
        'enrollment_date',
        'school_year',
        'status',
        'remarks',
        'approved_by',
        'approved_at',
    ];

    protected $casts = [
        'enrollment_date' => 'date',
        'approved_at'     => 'datetime',
    ];

    public function child()
    {
        return $this->belongsTo(Child::class);
    }

    public function approver()
    {
        return $this->belongsTo(User::class, 'approved_by');
    }
}
